schema/canonical/crc_test.go

```diff
@@ -1,24 +1,52 @@
nt\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Short\"},\"default\":0},{\"name\":\"segmentMessageRefNumber\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Short\"},\"default\":0},{\"name\":\"contentFilteringResult\",\"type\":{\"type\":\"enum\",\"name\":\"ContentFilteringResult\",\"symbols\":[\"BLOCKED\",\"ALLOWED\",\"NULL\"]},\"default\":\"ALLOWED\"},{\"name\":\"pdu\",\"type\":{\"type\":\"record\",\"name\":\"Pdu\",\"fields\":[{\"name\":\"name\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"},\"default\":\"\"},{\"name\":\"isRequest\",\"type\":\"boolean\",\"default\":false},{\"name\":\"commandLength\",\"type\":\"int\",\"default\":0},{\"name\":\"commandId\",\"type\":\"int\",\"default\":4},{\"name\":\"commandStatus\",\"type\":\"int\",\"default\":0},{\"name\":\"sequenceNumber\",\"type\":\"int\",\"default\":0},{\"name\":\"serviceType\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"},\"default\":\"\"},{\"name\":\"sourceAddress\",\"type\":{\"type\":\"record\",\"name\":\"Address\",\"fields\":[{\"name\":\"ton\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"npi\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"address\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"},\"default\":\"\"}]},\"default\":{}},{\"name\":\"destAddress\",\"type\":\"Address\",\"default\":{}},{\"name\":\"esmClass\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"protocolId\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"priority\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"scheduleDeliveryTime\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"},\"default\":\"\"},{\"name\":\"validityPeriod\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"},\"default\":\"\"},{\"name\":\"registeredDelivery\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"replaceIfPresent\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"dataCoding\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"defaultMsgId\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Byte\"},\"default\":0},{\"name\":\"shortMessage\",\"type\":{\"type\":\"bytes\",\"java-class\":\"[B\"},\"default\":\"ÿ\"},{\"name\":\"optionalParameters\",\"type\":{\"type\":\"array\",\"items\":{\"type\":\"record\",\"name\":\"Tlv\",\"fields\":[{\"name\":\"tag\",\"type\":{\"type\":\"int\",\"java-class\":\"java.lang.Short\"},\"default\":0},{\"name\":\"value\",\"type\":{\"type\":\"bytes\",\"java-class\":\"[B\"},\"default\":\"ÿ\"},{\"name\":\"tagName\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"},\"default\":\"\"}]},\"java-class\":\"java.util.ArrayList\"},\"default\":[]}]},\"default\":{}}],\"default\":{}}", "0c947f601de7ce84"},
```

Comment:
This test is just for the CRC64 computation, I don't think it makes sense to add testing the canonical form of a schema here.

Reply:
I Agree, I will separate the canonical from the crc and hardcode the canonical form in the crc test.

Reply:
reverted the tests